addons/account/models/account_move.py

```diff
@@ -545,8 +545,8 @@ def _onchange_amount_currency(self):
             amount = line.amount_currency
             if line.currency_id and line.currency_id != line.company_currency_id:
                 amount = self.currency_id.with_context(date=line.date).compute(amount, line.company_currency_id)
-            line.debit = amount > 0 and amount or 0.0
-            line.credit = amount < 0 and -amount or 0.0
+            line.debit = amount > 0 and amount or line.debit or 0.0
+            line.credit = amount < 0 and -amount or line.credit or 0.0
```

Comment:
These two lines are wrong. Doing this would mean you can have a line generated with a currency_id different from the company_currency, and have an amount_currency not matching the debit/credit of this line. Let's avoid that ;) You needed this fix because you created your account.move.line with too much data, I've explained that on the enterprise PR.